pkg/apiserver/debugapi/endpoint.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debugapi
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver/model"
+)
+
+var (
+	ErrPathParamNotMatched = ErrNS.NewType("path_param_not_matched")
+	ErrValueTransformed    = ErrNS.NewType("error_value_transformed")
+)
+
+type ValueGetter interface {
+	Get(key string) string
+}
+
+type EndpointAPI struct {
+	ID        string             `json:"id"`
+	Component model.NodeKind     `json:"component"`
+	Path      string             `json:"path"`
+	Method    string             `json:"method"`
+	Host      EndpointAPIParam   `json:"host"`
```

Comment:
Do we really need this host field for a model to record something? I think its value is not decided until user feeds a real host (when model is no longer a model).